src/Rule/Compare.php

```diff
@@ -24,28 +25,28 @@ abstract class Compare implements SerializableRuleInterface, SkipOnEmptyInterfac
 
     /**
      * Constant for specifying the comparison as string values.
-     * No conversion will be done before comparison.
+     * Values will be converted to strings before comparison.
      *
      * @see $type
      */
     public const TYPE_STRING = 'string';
     /**
      * Constant for specifying the comparison as numeric values.
-     * String values will be converted into numbers before comparison.
+     * Values will be converted to float numbers before comparison.
      *
      * @see $type
      */
     public const TYPE_NUMBER = 'number';
 
     private array $validOperators = [
-        '==' => 1,
```

Comment:
Accessing array by key is O(1). Checking for value presence is O(N). There aren't many operators so probably it doesn't worth optimising though.

Reply:
Yes, but considering the amount of operators, I think it's not worth having a map.